airflow/www/package.json

```diff
@@ -51,6 +51,8 @@
     "mini-css-extract-plugin": "^0.4.1",
     "moment-locales-webpack-plugin": "^1.2.0",
     "style-loader": "^0.21.0",
+    "stylelint": "^13.3.0",
```

Comment:
Are these required dependencies of our application or dependencies of the development environment?

Reply:
They are development environment dependencies (`devDependencies`).

Reply:
But you install it in a virtual environment provided by pre-commit. Do you also need it here the second time?

Reply:
Are developers running pre-commit in their development environment (I'm not. Is that a thing?)? (I'm not well versed in pre-commit.)

I suppose it isn't technically required—though as a developer, I want to be checking against these requirements before making any local commits, and utilize the package in CLI and in conjunction with my IDE plugin.

Reply:
Pre-commit downloads and installs the node  Most developers use pre-commit because it is checked in Travis. Without checking changes by pre-commit it cannot be accepted.
https://github.com/pre-commit/pre-commit/blob/master/pre_commit/languages/node.py#L59

I am only afraid that if you have the same app in two places, they may behave differently due to additional dependencies. The pre-commit should also integrate with the IDE because the IDE uses git, but I don't use git in the IDE.

Reply:
Generally, this frontend is not intensively developed and it is possible that we need more advanced solutions, e.g. installing the node in Breeze and then starting the stylelint by breeze. 

https://github.com/apache/airflow/blob/master/.pre-commit-config.yaml#L195-L200
https://github.com/apache/airflow/blob/master/.pre-commit-config.yaml#L223-L229
https://github.com/apache/airflow/blob/master/.pre-commit-config.yaml#L336-L362

However, I have no opinion on this matter. If you prefer to have packages in package.json, this will ok for me, but it's worth adding a comment that we should keep two places in sync.

Reply:
I would prefer to have them in package.json as well please.

Reply:
Thank you for the additional context @mik-laj. I've left them in both, but added a comment as you suggested.